src/motion_database.rs

```diff
@@ -0,0 +1,101 @@
+use bevy::prelude::*;
+use bvh_anim::{Frame, Joint};
+use serde::{Deserialize, Serialize};
+use std::{fs, io::Write};
+
+use crate::{
+    bvh::{bvh_asset::BvhAsset, bvh_player::get_pose},
+    ui::BuildConfig,
+};
+
+pub struct MotionDatabasePlugin;
+
+impl Plugin for MotionDatabasePlugin {
+    fn build(&self, _app: &mut App) {}
+}
+
+pub type Pose = Vec<Vec<f32>>;
+
+#[derive(Serialize, Deserialize)]
+pub struct TrajectoryPosition {
+    pub position: Vec3,
+    pub time: f32,
+}
+
+#[derive(Serialize, Deserialize, Default)]
+pub struct MotionData {
+    pub trajectories: Vec<TrajectoryPosition>,
+    pub trajectory_offsets: Vec<usize>,
+    pub poses: Vec<Pose>,
+    pub pose_offset: Vec<usize>,
+}
+
+// trajectories: [ bvh0:traj0, bvh0:traj1, bvh0:traj2, bvh1:traj0, bvh1:traj1, bvh2:traj0 ]
+//      offsets: [ 0, 3, 5, 6 ]
+
+pub fn extract_motion_data(bvh_asset: &Assets<BvhAsset>, build_config: &mut BuildConfig) {
+    let mut motion_data = MotionData::default();
```

Comment:
```suggestion
    let mut motion_data = MotionData::default();
    let mut trajectory_data_len: usize = 0;
    let mut motion_data_len: usize = 0;
```

Reply:
move `trajectory_data_len` and `motion_data_len` here so that the accumulation continues throughout the loop.